### PR TITLE
Added tests for chat

### PR DESCRIPTION
**Pull request by crookedneighbor**
Started the process of writing tests for the chat methods in api.

Far from done (or perfect), but it's a start. 

Anyone want to look over these and suggest corrections/improvements?

### CODE REVIEW

File: test/api.mocha.coffee [hunk 1]
```diff
@@ -172,6 +172,87 @@ describe "API", ->
...
+              message: "Test MSG"
+            ).end (res) ->
+              expectCode res, 200
+              chat = res.body.message
+              expect(chat.id).to.be.ok
+              expect(chat.timestamp).to.be.ok
+              expect(chat.likes).to.be.empty
+              expect(chat.flags).to.be.empty
+              expect(chat.flagCount).to.be 0
+              expect(chat.uuid).to.be.ok
```
Comment (crookedneighbor) [inline test/api.mocha.coffee]: Now that I look at it here, this and the following couple of lines should probably compare to the user object.

**Comment by Alys** [comment]
@crookedneighbor Just checking: Are these still relevant? They look useful at first glance, but I thought I'd ask before poking around further.

**Comment by crookedneighbor** [comment]
They should be. I'll try to rebase this branch over the weekend and merge it in.

**Comment by Alys** [comment]
Don't feel forced to if you have other things to do! We know you're busy at the moment.

**Comment by gisikw** [comment]
Just did a rebase and these tests still pass and are awesome :)

**Comment by crookedneighbor** [comment]
@gisikw if you already have a branch with these commits in them, feel free to PR it! Should close this one automatically if it gets merged in.

**Comment by Alys** [comment]
@gisikw :heart:

**Comment by gisikw** [comment]
Ah, cool cool. Lemme pull this and your Todos branch into a shared test update branch, and I'll PR it :)

**Comment by Alys** [comment]
closed by https://github.com/HabitRPG/habitrpg/pull/4931 thanks to @gisikw

**Pull request closed without merging**